compiler/modules/nback/build.bal

```diff
@@ -1106,48 +1163,67 @@ final readonly & map<bir:OrderOp> flippedOrderOps = {
-    builder.condBr(eitherNil, eitherNilBB, neitherNilBB);
-
-    builder.positionAtEnd(eitherNilBB);
-    llvm:Value bothNil = builder.iBitwise("and", lhsIsNil, rhsIsNil);
+    bir:OrderType orderTy = insn.orderType;
+    llvm:Value? compareResult = ();
+    if orderTy is bir:OptOrderType {
+        match orderTy.opt {
+            t:UT_INT => {
+                compareResult = builder.call(buildRuntimeFunctionDecl(scaffold, intCompareFunction), [lhs, rhs]);
+            }
+            t:UT_FLOAT => {
+                compareResult = builder.call(buildRuntimeFunctionDecl(scaffold, floatCompareFunction), [lhs, rhs]);
+            }
+            t:UT_BOOLEAN => {
+                compareResult = builder.call(buildRuntimeFunctionDecl(scaffold, intCompareFunction), [lhs, rhs]);
+            }
+            t:UT_STRING => {
+                compareResult = builder.call(buildRuntimeFunctionDecl(scaffold, stringCompareFunction), [lhs, rhs]);
+            }
+        }
 
-    if insn.op is "<=" || insn.op is ">=" {
-        buildStoreBoolean(builder, scaffold, bothNil, insn.result);
-    }
-    else {
-        buildStoreBoolean(builder, scaffold, llvm:constInt(LLVM_BOOLEAN, 0), insn.result);
     }
-    builder.br(joinBB);
-
-    builder.positionAtEnd(neitherNilBB);
-    bir:OptOrderType orderTy = <bir:OptOrderType> insn.orderType;
-    match orderTy.opt {
-        t:UT_INT => {
-            llvm:Value lhsUntagged = buildUntagInt(builder, scaffold, <llvm:PointerValue>lhs);
-            llvm:Value rhsUntagged = buildUntagInt(builder, scaffold, <llvm:PointerValue>rhs);
-            buildCompareInt(builder, scaffold, buildIntCompareOp(insn.op), lhsUntagged, rhsUntagged, result);
+    else if orderTy is bir:ArrayOrderType {
+        match orderTy[0].opt {
+            t:UT_INT => {
+                compareResult = builder.call(buildRuntimeFunctionDecl(scaffold, arrayIntCompareFunction), [lhs, rhs]);
+            }
+            t:UT_FLOAT => {
+                compareResult = builder.call(buildRuntimeFunctionDecl(scaffold, arrayFloatCompareFunction), [lhs, rhs]);
+            }
+            t:UT_BOOLEAN => {
+                compareResult = builder.call(buildRuntimeFunctionDecl(scaffold, arrayIntCompareFunction), [lhs, rhs]);
+            }
+            t:UT_STRING => {
+                compareResult = builder.call(buildRuntimeFunctionDecl(scaffold, arrayStringCompareFunction), [lhs, rhs]);
+            }
         }
-        t:UT_FLOAT => {
-            llvm:Value lhsUntagged = buildUntagFloat(builder, scaffold, <llvm:PointerValue>lhs);
-            llvm:Value rhsUntagged = buildUntagFloat(builder, scaffold, <llvm:PointerValue>rhs);
-            buildCompareFloat(builder, scaffold, buildFloatCompareOp(insn.op), lhsUntagged, rhsUntagged, result);
+    }
+    if compareResult is () {
+        panic error("Failed to find runtime compare function");
+    }
+    else {
+        llvm:Value? resultValue = ();
+        match insn.op {
+            ">=" => {
```

Comment:
Have a map that maps e.g. ">=" to "sge" (don't we have one already), and then don't repeat the statement.

Reply:
Fixed by adding a new table to record both predicate and expected value